tests/lit-tests/float16_interface.ispc

```diff
@@ -0,0 +1,6 @@
+// RUN: %{ispc} %s --target=host -h %t.h
+// RUN: FileCheck --input-file=%t.h %s
+
+// CHECK: extern void export_float16(__fp16 arg0, __fp16 * arg1);
```

Comment:
Now as we have test for header files, I see that it needs to be customized for different compilers.

`clang` and `gcc` support `__fp16`, while MSVC seems to not. I guess we need to create and typedef, which would map to different underlying types depending on the host compiler. For MSVC `uint16_t` seem to be the safe choice. They map to `uint16_t` for DirectX as well: https://docs.microsoft.com/en-us/windows/win32/dxmath/half-data-type

This change may go as a separate fix.

Reply:
Yep. Will add this as a separate commit.